aiida/orm/nodes/data/code/containerized.py

```diff
@@ -0,0 +1,167 @@
+        """
+        return self.base.attributes.get(self._KEY_ATTRIBUTE_ESCAPE_EXEC_LINE)
+
+    @escape_exec_line.setter
+    def escape_exec_line(self, value: bool) -> None:
+        """Set whether to escape execute line in double quotes
+
+        :param value: True for escape execute line in double quotes, False for not escape.
+        """
+        type_check(value, bool)
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_ESCAPE_EXEC_LINE, value)
+
+    @property
+    def image(self) -> str:
+        """The image of container
+
+        :return: The image of container.
+        """
+        return self.base.attributes.get(self._KEY_ATTRIBUTE_IMAGE)
+
+    @image.setter
+    def image(self, value: str) -> None:
+        """Set the image name of container
+
+        :param value: The image name of container.
+        """
+        type_check(value, str)
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_IMAGE, value)
+
+    def get_engine_command(self) -> str:
+        """Return the engine command with image field set with image name.
+
+        :return: The engine command in script job with image field set with the image name.
+        """
+        cmdline = self.engine_command.format(image=self.image)
+
+        return cmdline.split()
+
+    @classmethod
+    def _get_cli_options(cls) -> dict:
+        """Return the CLI options that would allow to create an instance of this class."""
+        options = {
+            'engine_command': {
+                'required': True,
+                'prompt': 'Engine command',
+                'help': 'The command to run container must contain {image} for image.',
+                'type': click.STRING,
+            },
+            'image': {
+                'required': True,
+                'type': click.STRING,
+                'prompt': 'Image',
+                'help': 'Image of the container to run executable.',
+            },
+            'escape_exec_line': {
+                'type': click.BOOL,
+                'default': True,
+                'prompt': 'Escape whole executable line?',
+                'help': 'True to escape executable line in double quotes, only required by docker.',
+            },
+        }
+        options.update(**super()._get_cli_options())
+
+        return options
+
+
+class InstalledContainerizedCode(Containerized, InstalledCode):
+    """Data plugin representing an executable code in container on a remote computer."""
+
+    def __init__(self, **kwargs):
+        """Construct a new instance.
+        """
+        super().__init__(**kwargs)
+
+    @property
+    def full_label(self) -> str:
+        """Return the full label of this code.
+
+        :return: The full label of the code.
+        """
+        return f'{self.label}@{self.computer.label}@containerized'
+
+
+class PortableContainerizedCode(Containerized, PortableCode):
+    """Data plugin representing an a container and executable code stored in AiiDA's storage."""
```

Comment:
```suggestion
    """Data plugin representing a container and executable code stored in AiiDA's storage."""
```
Question: the docstring seems to suggest that the container itself is also stored in AiiDA's storage, but that is not the case right? If so, maybe the following would be better
> Data plugin representing an executable code stored in AiiDA's storage to be run in a container.